src/ucp/wireup/select.c

```diff
@@ -1725,13 +1725,15 @@ ucp_wireup_add_am_bw_lanes(const ucp_wireup_select_params_t *select_params,
             /* do not continue searching since we found AM lane (and there is
              * only one AM lane) */
             am_lane = lane_desc_idx;
+            select_ctx->lane_descs[lane_desc_idx].path_index = 0;
```

Comment:
why needed?

Reply:
In case when `am_lane` has undefined `path_index`, it would be merged with one of `am_bw_lanes`

Reply:
ok, why do we need to fix that?
also this change breaks wire compatibility, need to add some protection to avoid that

Reply:
In test case where we have `IB_NUM_PATHS=16` and `MAX_EAGER_LANES=16` without this change UCX first will select 1 lane for AM, then will select 15 lanes for AM_BW, but since AM lane has undefined path_index, it would be merged with one of selected AM_BW lanes, so the resulting number of lanes would be 15.

Reply:
i'm afraid this would affect other use-case, like selecting other type of lanes (e.g. amo or rma). With this code they ma not be merged with AM lane anymore, which is not good in terms of resources consumption